docs/variants/hardkernel_odroid_h4/recovery.md

```diff
@@ -0,0 +1,92 @@
+# Recovery
+
+## Intro
+
+The following documentation describes the process of recovering hardware from
+the brick state using an [RTE](../../transparent-validation/rte/introduction.md)
+and Dasharo open-source firmware.
+
+## Dual BIOS
+
+The **ODROID H4+ and ODROID-H4 Ultra** devices both have the Dual BIOS
+feature. If the device does not boot properly, the backup flash chip can be
+selected using the flash selector jumper.
+
+![](../../images/odroid_flash_switch.jpg)
+*First bios chip selected using the flash selector*
+
+1. Move the flash selector jumper to select the backup flash chip. If the
+   device does not boot from any of the flash chips, an external recovery
+   needs to be performed. In that case continue with the recovery instructions
+   for the ODROID H4 model.
+2. If the device boots, boot to an OS.
+3. Move the flash selector jumper back to it's original position selecting the
+   previous flash chip, from which the device does not boot.
+4. With the device working, an OS booted and the flash chip selector moved
+   back, the firmware can be flashed using the command:
+
+    ```bash
+    flashrom -p internal -w [path_to_binary]
+    ```
```

Comment:
@philipandag, did you test that scenario, or are you just writing theoretically?

Reply:
Tested it with original BIOS from https://dn.odroid.com/ODROID-H4/bios and it worked

```
bash-5.2# flashrom -p internal -w ADLN-H4_B1.5.bin
flashrom v1.2-1037-g5b4a5b40 on Linux 6.6.21-yocto-standard (x86_64)
flashrom is free software, get the source code at https://flashrom.org

Using clock_gettime for delay loops (clk_id: 1, resolution: 1ns).
No DMI table found.
Found chipset "Intel Alder Lake-N".
Enabling flash write... SPI Configuration is locked down.
FREG0: Flash Descriptor region (0x00000000-0x00000fff) is read-write.
FREG1: BIOS region (0x00600000-0x00ffffff) is read-write.
FREG2: Management Engine region (0x00001000-0x00413fff) is read-write.
Enabling hardware sequencing because some important opcode is locked.
OK.
Found Programmer flash chip "Opaque flash chip" (16384 kB, Programmer-specific) on internal.
Reading old flash chip contents... done.
Erasing and writing flash chip... Erase/write done.
Verifying flash... VERIFIED.
```